src/pages/signIn/signUp.jsx

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios';
+import { Formik } from 'formik';
+import React, { useState } from 'react';
+import { useQuery } from 'react-query';
+import { useDispatch } from 'react-redux';
+import * as Yup from 'yup';
+
+import { serverPath } from '../../constants/noteList';
+import { getUserInfo } from '../../redux/actions/actionCreators';
+import { SignForm, ValidContainer, StyledField, ErrorTip, SignInButton, RegularText, StyledLink, StyledTitle } from './styled';
+
+const ValidScheme = Yup.object().shape({
+    email: Yup.string().email('Please use @ and . for adding email').required('Please enter Email'),
+    password: Yup.string()
+        .min(3, 'password must contain at least 3 symblos')
+        .required('Please enter password'),
+    firstName: Yup.string().required('Please enter your name'),
+    lastName: Yup.string().required('Enter your last name'),
+    dateOfBirth: Yup.string().required('Please pick date of birth'),
+    confirmPassword: Yup.string().oneOf([Yup.ref('password'),null],'Passwords must match ')
+});
+const SignUp = () => {
+    const [newUser, setNewUser] = useState(null);
+    const dispatch=useDispatch()
+    const sendDataToServer = async (data) => {
+        const resp = await axios.post(serverPath, data);
+        dispatch(getUserInfo(data))
+        setNewUser(null)
+        return resp.status;
+    };
+    useQuery(
+        'addUser',
+        async () => {
+            sendDataToServer(newUser);
+        },
+        {
+            enabled: !!newUser,
+        }
+    );
+    return (
+        <Formik
+            initialValues={{
+                email: '',
+                password: '',
+                lastName: '',
+                firstName: '',
+                dateOfBirth: '',
+                confirmPassword:''
+            }}
+            validationSchema={ValidScheme}
+            onSubmit={async (values, actions) => {
+                const dataToSend = {
+                    firstName: values.firstName,
+                    lastName: values.lastName,
+                    dateOfBirth: values.dateOfBirth,
+                    password: values.password,
+                    email: values.email,
+                };
+                setNewUser(dataToSend);
+                actions.resetForm();
+            }}
+        >
+            {({ errors, touched }) => (
+                <SignForm>
+                    <StyledTitle>sign up</StyledTitle>
+                    <ValidContainer>
+                        <label htmlFor="firstName">Please Enter First namr</label>
+                        <StyledField id="firstName" name="firstName" placeholder="John" />
+                        {errors.firstName && touched.firstName &&(
+                            <ErrorTip>{errors.firstName}</ErrorTip>
+                        )}
+                    </ValidContainer>
+                    <ValidContainer>
+                        <label htmlFor="lastName">Please Enter last name</label>
```

Comment:
lots of duplicated code, create a formik text input, and use it in all places where u want, just provide different names/labels etc.